op-signer.md

```diff
@@ -0,0 +1,253 @@
+
+### Get Service Account Key File
+
+Enter the service account detail page and click the Keys tab.
+
+- Click `Add Key / Create new key`.
+- For Key type, select the JSON key option, then click Create. The file will automatically download to your computer.
+- Place the *.json file you just downloaded in a directory of your choosing for later use.
+
+
+## Create a KMS Key
+
+The following steps prepare you to create a GCP KMS key.
+
+### Enable Access to KMS API
+
+If not done yet, go to https://console.cloud.google.com/flows/enableapi?apiid=cloudkms.googleapis.com, confirm that the project is correct, and enable the `Cloud Key Management Service (KMS) API`.
+
+### Create a Key Ring
+
+A key ring organizes keys in a specific Google Cloud location and allows you to manage access control on groups of keys. Go to https://console.cloud.google.com/security/kms, where you may see key rings already created. Optionally, you can create new key rings by clicking the `CREATE KEY RING` button and assigning the following values as an example:
+
+```
+key-ring-name: op-signer
+Multi-region: Global
+```
+
+Then click `CREATE`.
+
+### Create a Key
+
+Choose a key ring on the page https://console.cloud.google.com/security/kms and click `CREATE KEY`. In the `Create Key` page, assign values as in the following example and click `CONTINUE` for each step:
+
+```
+key name: op-challenger (Depending on the consumer of the key)
+Protection Level: HSM (Protection level must be set to HSM to use secp256k1)
+Key material: HSM-generated key
+Purpose: Asymmetric sign
+Algorithm: Elliptic Curve secp256k1 - SHA256 Digest (Ethereum compatible)
+Default expiration: 30 days
+```
+
+Then click `CREATE`.
+
+### Grant Permissions
+
+Click the `Permissions` tab of the Key detail page and click `Grant Access`.
+
+- Add the email address of the service account as principals from [this step](#create-a-service-account) and assign proper roles. For example:
+
+```
+Add principals: swc-signer-sa@signing-test-450710.iam.gserviceaccount.com
+Assign roles: roles/cloudkms.signerVerifier
+```
+
+Note that only IAM principals with Owner (roles/owner) or Cloud KMS Admin (roles/cloudkms.admin) roles can grant or revoke access to Cloud KMS resources.
+
+### Get Resource Name of the Key
+
+Click on the created item and in the `Versions` tab of the key detail page, click the button under `Actions` and select `Copy resource name`.
+
+For example:
+
+```
+projects/swc-signer/locations/global/keyRings/op-signer/cryptoKeys/op-challenger/cryptoKeyVersions/1
+```
+
+You will add this value as `key` in `op-signer/config.yaml` in the `infra` repo.
+
+## Deploy op-signer Service
+
+### Build the Source
+
+Log on to the server where the op-signer service will be deployed.
+
+```bash
+git clone https://github.com/QuarkChain/infra.git
+cd infra/op-signer
```

Comment:
It is the op-signer with `address` command support?

Reply:
Yes, after https://github.com/QuarkChain/infra/pull/1 merged to main branch.